compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -249,6 +266,11 @@ static void addDispatchRegionCreationPasses(OpPassManager &passManager) {
       // handle explicit captures as materialized as dispatch workgroup operands
       // and block arguments.
       .addPass(IREE::Flow::createCloneProducersIntoDispatchRegionsPass)
+      .addPredicatedPass(clEnableDataTiling,
```

Comment:
You maybe want to move this "after" the collapse dimensions pass.

Reply:
I intuitively had thought that it would be better to handle encodings while higher dimensionality is still present (since we essentially make fusion decisions again after the encoding ops are hoisted out of dispatch regions). Is there a reason why it might be better to have it after collapse dims?

Reply:
The collapse dimensions are to fold away dimensions we never really should have to worry about after fusion. These are dimensions that are contiguous in the range of the indexing maps for all operands, and are either all parallel or all reduction. So these dimensions should not make any difference for code-generation which is what set encoding is for.

Reply:
The reason I want the extra dimensions still is because I reuse the fusion logic to decide where to place encodings after hoisting them out of the dispatches: https://github.com/iree-org/iree/pull/18069